app/models.py

```diff
@@ -0,0 +1,607 @@
+
+"""
+ this defines the table that holds the relationship between a User Model 
+ and the Url model
+"""
+user_url = db.Table('user_url',
+                    db.Column('user_id', db.Integer,
+                              db.ForeignKey('users.id')),
+                    db.Column('urls_id', db.Integer,
+                              db.ForeignKey('url.id')))
+
+
+class User(db.Model, UserMixin):
+    """
+    This class defines the user table schema and also handles user object related
+    functionalities.
+    """
+    __tablename__ = 'users'
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(20), unique=True, index=True,
+                         nullable=False)
+    lastname = db.Column(db.String(20), nullable=False)
+    firstname = db.Column(db.String(20), nullable=False)
+    email = db.Column(db.String(30), unique=True, index=True, nullable=False)
+    password_hash = db.Column(db.String(128))
+    is_admin = db.Column(db.Boolean, default=False)
+    date_added = db.Column(db.DateTime, default=datetime.utcnow())
+    url = db.relationship("Url", secondary=user_url, backref='users',
+                          lazy='dynamic')
+    short_url = db.relationship("ShortenUrl",  backref='users',
+                                lazy='dynamic')
+    token = db.relationship("Token", backref="users", lazy="dynamic")
+
+    @property
+    def password(self):
+        raise AttributeError("Password is not a readable attribute")
+
+    @password.setter
+    def password(self, password):
+        """
+        sets the password attribute , assigning the argument passed in as
+        the new value
+        :param password: 
+        :return: 
+        """
+        self.password_hash = generate_password_hash(password)
+
+    @property
+    def url_list(self):
+        """
+        returns a list of all the long urls belonging to a 
+        particular user
+        :return list_of_long_urls: 
+        """
+        return self.url
+
+    @property
+    def short_url_list(self):
+        """
+        returns a list of all the long urls belonging to a 
+        particular user
+        :return list_of_shortened_urls: 
+        """
+        return self.short_url
+
+    def verify_password(self, password):
+        """
+        verifies that the password passed in as argument matches the
+        one saved in the database
+        :param password: 
+        :return boolean(True if it matches else False): 
+        """
+        return check_password_hash(self.password_hash, password)
+
+    def generate_auth_token(self, expiration):
+        """
+        generates a token using the serializer to hash the user's id 
+        and expiration time with the secret key
+        :param expiration: 
+        :return token (str): 
+        """
+        s = Serializer(current_app.config['SECRET_KEY'],
+                       expires_in=expiration)
+        return [s.now(), s.dumps({'id': self.id}).decode('ascii')]
```

Comment:
I think it would be helpful to your users to perform these two functions on lines preceding this and also give them descriptive variable names. You may even consider returning a dictionary. I noticed I had to think briefly as to what was in index 1 of this list since it wasn't explicit. Though the variable name there helped as well.